packages/headless/src/ssr-next/commerce/controller-utils.test.ts

```diff
@@ -0,0 +1,276 @@
+      buildControllersWithDefaultSetup();
+      expect(ControllerBuilder).toHaveBeenNthCalledWith(
+        1,
+        {
+          build: expect.any(Function),
+          listing: true,
+          search: true,
+          standalone: true,
+          recommendation: true,
+        },
+        mockEngine,
+        undefined
+      );
+    });
+
+    it('should call #ControllerBuilder for the controller with props with the correct arguments', () => {
+      buildControllersWithDefaultSetup();
+      expect(ControllerBuilder).toHaveBeenNthCalledWith(
+        2,
+        {
+          buildWithProps: expect.any(Function),
+          listing: true,
+          search: true,
+          standalone: true,
+          recommendation: true,
+        },
+        mockEngine,
+        {
+          initialState: {
+            prop1: 'value1',
+            prop2: 42,
+          },
+        }
+      );
+    });
+
+    describe('when the definition contains a mix of solution type controllers', () => {
+      let controller1: ControllerDefinition<Controller>;
+      let controller2: ControllerDefinition<Controller>;
+
+      beforeEach(() => {
+        controller1 = defineMockCommerceController({
+          search: true,
+          recommendation: false,
+        });
+        controller2 = defineMockCommerceController({
+          search: false,
+          recommendation: false,
+        });
+      });
+
+      it('should call #ControllerBuilder for the controllers with the appropriate solution type', () => {
+        buildControllerDefinitions({
+          definitionsMap: {controller1, controller2},
+          engine: mockEngine,
+          solutionType: SolutionType.search,
+        });
+
+        // should only call for controller1 with listing and search
+        expect(ControllerBuilder).toHaveBeenCalledTimes(1);
+      });
+
+      it('should call #ControllerBuilder with the appropriate arguments', () => {
+        buildControllerDefinitions({
+          definitionsMap: {controller1, controller2},
+          engine: mockEngine,
+          solutionType: SolutionType.search,
+        });
+
+        expect(ControllerBuilder).toHaveBeenCalledWith(
+          expect.objectContaining({
+            build: expect.any(Function),
+            search: true,
+          }),
+          mockEngine,
+          undefined
+        );
+      });
+
+      it('should not call #ControllerBuilder if no controller is defined for the solution type', () => {
+        controller1 = defineMockCommerceController({
+          search: false,
+        });
+        controller2 = defineMockCommerceController({
+          search: false,
+        });
+
+        buildControllerDefinitions({
+          definitionsMap: {controller1, controller2},
+          engine: mockEngine,
+          solutionType: SolutionType.search,
+        });
+
+        expect(ControllerBuilder).not.toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('#createStaticState', () => {
```

Comment:
You could define the mockSearchActions in this `describe` block's `beforeEach`, instead of in the top `describe`'s.